js/memory.js

```diff
@@ -1,9 +1,27 @@
 class MemoryGame {
-  constructor(cards){
+  constructor(cards) {
     this.cards = cards;
-    // add the rest of the class properties here
+    // this.pickedCards = pickedCards;
+    // this.pairsClicked = pairsClicked;
+    // this.pairsGuessed = pairsGuessed;
+  }
+  shuffleCards(cardsObj) {
+    if (!cardsObj) {
+      return undefined;
+    }
+    for (let i = 0; i < cardsObj.length; i++) {
+      let rand = Math.floor(Math.random() * cardsObj.length);
+      cardsObj.push(cardsObj[rand]);
+      cardsObj.splice(rand, 1);
+    }
   }
-  shuffleCards() {}
   checkIfPair(card1, card2) {}
   isFinished() {}
-}
\ No newline at end of file
+}
+
+let cardsArrTemp = cardsArr;
+for (let i = 0; i < cardsArrTemp.length; i++) {
```

Comment:
If this is double code, I would remove it.

Reply:
Thanks, yes, double-code now removed :)